File: Dynamic_Programming/20_Oct_2019_binomial_coefficient/20_Oct_2019_binomial_coefficient.cpp
//naive binomial coefficient

#include<bits/stdc++.h>
using namespace std;

int binomialCoeff(int n, int k){
    //BASE CASE
    if(k == 0 || k == n){
        return 1;
    }

    //recur
    return binomialCoeff(n-1, k-1) + binomialCoeff(n-1, k);

}

//driver code
int main(){
    int n = 5, k = 2;

    cout << "Value of C("<<n<<", "<<k<<") is "<< binomialCoeff(n,k);
    return 0;
}

File: Dynamic_Programming/october/20_Oct_2019_binomial_coefficient/20_Oct_2019_binomial_coefficient.py
#naive binomialcoeff
def binomialCoeff(n,k):
    
    #base case
    if k == 0 or k == n:
        return 1
    
    #Recursion
    return binomialCoeff(n-1,k-1) + binomialCoeff(n-1,k)

#driver code
n = 5
k = 2

print("The binomial coeff of {} and {} is {}".format(n,k,binomialCoeff(n,k)))